reflex/.templates/web/utils/state.js

```diff
@@ -161,8 +161,14 @@ export const applyEvent = async (event, socket) => {
   if (event.name == "_download") {
     const a = document.createElement("a");
     a.hidden = true;
-    // Special case when linking to uploaded files
-    a.href = event.payload.url.replace("${getBackendURL(env.UPLOAD)}", getBackendURL(env.UPLOAD))
+    // Special case when linking to uploaded files (when rx.download is returned from upload_files handler())
+    a.href = event.payload.url.replace(
+      "${getBackendURL(env.UPLOAD)}",
+      getBackendURL(env.UPLOAD)
+    );
+    if (event.payload.filename) {
+      a.href = a.href + "?filename=" + event.payload.filename;
```

Comment:
potentially minor issue, but is there a case where the href might already have query params?

Reply:
I don't think so.